### DISCUSSION TITLE
Reversible Retraction and Samplers

### DISCUSSION THREAD

**Discussion by tolgabirdal** [geoopt/geoopt#174] [General]
I am wondering why geoopt requires retraction to be 'reversible' (not sure what this refers to) for samplers to work. Why wouldn't, for example, SGRLD work on a sphere? The sphere manifold (whose retraction is apparently not 'reversible') is one of the typical examples people give when illustrating dynamics-based MCMC methods on manifolds. See here: https://papers.nips.cc/paper/2016/file/77f959f119f4fb2321e9ce801e2f5163-Paper.pdf

It would be good to have some elaboration on such implementation choice.

**Comment by AlexanderZeilmann** [comment]
Regarding the definition of 'reversible': https://github.com/geoopt/geoopt/discussions/159

**Comment by ferrine** [comment]
Reversibility is a standard wish for developing an MCMC kernel. Is link might be helpful https://stats.stackexchange.com/questions/226550/reversibility-in-mcmc
CC @justanothercoder who implemented MCMC in geoopt

**Comment by ferrine** [comment]
@tolgabirdal I moved the issue to a discussion place
